### PR TITLE
Add standardized debug logging

### PR DESCRIPTION
**Pull request by david-christiansen**
Now, methods have an effect available to use for printf-debugging.

### CODE REVIEW

**Comment by david-christiansen** [comment]
In particular, this beats out `Debug.Trace` because we can arrange for it to go to the right place, which can differ based on how the app is served.

File: argo/src/Argo/Socket.hs [hunk 1]
```diff
@@ -84,15 +85,15 @@ startListening addr =
...
+acceptClient :: App s -> N.Socket -> IO ()
+acceptClient app s =
 
   do (c, peer) <- N.accept s
      h         <- N.socketToHandle c ReadWriteMode
      -- don't use c after this, it is owned by h
 
      log ("CONNECT: " ++ show peer)
-     forkFinally (serveHandlesNS logH h h app) $ \res ->
+     forkFinally (serveHandlesNS (Just stderr) h h app) $ \res ->
```
Comment (kwf) [inline argo/src/Argo/Socket.hs]: Do we want to hard-code the use of `stderr` here? For now it seems fine, and perhaps the simplicity of making the decision in this location warrants the (slight) loss of generality as compared to letting the caller specify the handle. What do you think?

**Comment by kwf** [comment]
Except for the question above, everything here looks good to me—this is how I would have implemented this feature as well.

File: argo/src/Argo/Socket.hs (hunk 1 above)
Comment (david-christiansen) [reply argo/src/Argo/Socket.hs]: I thought about it the way you did. I went with simplicity when I couldn't figure out any situation where I'd want something else. If such a situation arises, we can always add the parameter.

**Comment by david-christiansen** [comment]
Thanks! Mind if I merge, given that answer?

**Comment by kwf** [comment]
Sounds good to me!

**Comment by kwf** [comment]
(and so it is done)

**Pull request merged**